sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1396,6 +1396,16 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
       compressed = false,
       properties = rowStorage.properties ++ fileStorage.properties)
 
-    (ctx.LOCAL != null, storage, Some(DDLUtils.HIVE_PROVIDER))
+    val fileFormat = extractFileFormat(fileStorage.serde)
+    (ctx.LOCAL != null, storage, Some(fileFormat))
+  }
+
+  private def extractFileFormat(serde: Option[String]): String = {
```

Comment:
I think @maropu has kind of its point. Why don't you just use `STORED AS file_format` instead? BTW, I think it should be in `RelationConversions` with `spark.sql.hive.convertMetastoreXXX`.

Reply:
Yea, I think so... the third param is mainly used for data source tables. In this hive path, this param is just used for a flag, a hive table or not.

Reply:
@HyukjinKwon I used `STORED AS file_format` only.. if we use `STORED AS` it will be hive managed, so in this PR i  convert it to data source if convertible, we follow same behavior  in case of `CTAS` which is fixed in ` SPARK-25271`

Reply:
@Udbhav30, I more meant this seems not the right place to replace. You should add a configuration, and such replacement should be done in analysis/optimizing, not in the parser.

If that's the case, then why don't you use `USING file_format` explicitly? Can you please describe what this PR target to fix clearly?

Reply:
@HyukjinKwon, i can use `USING file_format` explicitly that would serve the purpose but i thought it is better to fix this and make it inline with `CTAS` behavior  which is fixed in this [PR](https://github.com/apache/spark/pull/22514)

If you agree to go ahead i can try making changes in analysis/optimizing layer instead of parser as suggested by you. 
Sure i will follow the template and update the PR description. thanks :)